src/main/java/io/cryostat/targets/TargetConnectionManager.java

```diff
@@ -85,6 +86,19 @@ public class TargetConnectionManager {
     private final Duration failedBackoff;
     private final Duration failedTimeout;
 
+    @ConfigProperty(name = ConfigProperties.URI_RANGE, defaultValue = "PUBLIC")
+    String uriRangeSetting;
+
+    public void connectToTarget(String urlString) throws Exception {
```

Comment:
I think this is not called anywhere.

Rather than placing the `range.validate(uri)` check here inside of `TargetConnectionManager`, I think it should be in `CustomDiscovery` (custom targets) and `Discovery` (discovery plugin registration/publication). This way the targets get rejected before they are ever stored, rather than before something tries to initiate a connection to them.

Or even inside `Target` in the pre-Persist hook instead, which would really guarantee that targets with out-of-range URIs cannot be stored. However, doing it here would also block the built-in discovery mechanisms from being able to discovery such targets, which I don't think is really what we want - it seems to me like this should only apply to custom targets and discovery plugins.